### DISCUSSION TITLE
January release - New rich text editor, JSON data type, and more

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#4527] [Announcements]
Hello Budis!

For the January release, we delivered some frequently requested features. Check them out below:

### JSON Data Type
The new JSON datatype in the Budibase internal DB brings much more flexibility to your data modeling in Budibase. JSON fields come with their own schema and can be used to create flexible data structures. The JSON datatype is excellent for forms, data entry, and storing responses from API requests.
[View documentation](https://docs.budibase.com/building-apps/data/data-types/json)

### Rich Text Component
We have implemented a completely new Rich Text component that will supercharge your writing experience with markdown. Features include live split-screen markdown preview, full-screen editing, and link and image support. The Rich Text component is also themable and works on all devices. We've also added a new Markdown Viewer component that allows you to view the generated markdown.
![rich-text-editor](https://user-images.githubusercontent.com/49767913/154275567-863d6434-0b2e-449f-85b9-35ccc9bc51c3.gif)

### Filtering on Relationships ( Postgres, MySQL, SQL server only)
The January release brings the long-awaited feature of being able to filter on relationships in SQL databases. Filter on your keys from relationships when using Postgres, MySQL, and SQL Server database.

### New tag component
Enrich the design and visualization of your apps with tags. Tags are a great way to categorize your content, and with the Budibase tags, you can update their colors, change their size, and add a close button (with actions!).
<img width="903" alt="CleanShot 2022-02-15 at 15 57 13@2x" src="https://user-images.githubusercontent.com/49767913/154275518-5d68b8a4-f931-4692-884c-a6971c9e95e5.png">

### Custom app URLs paths
Previously, Budibase URLs were a little… ugly. We prettified the URLs and they now look a lot better. You can also change the URL path of your app.

### Additional, important improvements
- Button actions now expose an output context, to use in the following steps. For example, you may have an action to save a row, then another action to navigate to that recently saved row's specific screen.
- A number of performance improvements - Budibase and your apps should feel a lot faster.

### Community shoutouts
A huge thank you to [Maurits Lourens](https://github.com/mslourens), [Hwansoo Kim](https://github.com/pueding), [Bastiaan Terhorst](https://github.com/bastiaanterhorst), and [SucculentAnimal](https://github.com/SucculentAnimal) for their contributions to Budibase over this last month ❤️

### Coming up in the February Release
- Official Google sheets integration (we promise!)
- Budibase API
- An updated table component

### We reached the 10,000 Github stars milestone
Recently, the Budibase Github repository was starred for its 10,000th time. This is an incredible achievement, and I'd like to thank our community for all of their support.
![tumblr_mfg58qXVNA1r3ty02o1_500](https://user-images.githubusercontent.com/49767913/154276091-da7309c1-6012-4053-bdb4-5256ebd1ff0f.gif)

**Comment by samg666** [comment]
Great team, great works! 

BTW, may I know from which version will has these new features?

**Reply by joebudi** [reply]
v1.0.67 will have these features.

**Comment by cbeulke** [comment]
Yeah, JSON data type and the button output context are two things I are absolutely happy to see! Especially the button output is something I hoped so much to happen soon... Thank you, you are doing great work!!!

**Comment by Shyru** [comment]
Congrats, very nice! 🎉 I'm especially happy about the rich text editor as I hacked something together with pseudo markdown-support and inline javascript. Having a real rich-text-editor will be so much better for our applicant tracking system!

**Comment by tmdoit** [comment]
Am I right that in the chat message with new release information you wrote "Tab" instead of "Tag"?

**Reply by joebudi** [reply]
Apologies for this. That was an unfortunate mistype. It is indeed Tag. The chat message has been updated!

**Comment by hendrikjap** [comment]
Awesome guys!

**Comment by oswaldB** [comment]
Hello,

very nice;

How to change the url?

Thank you

**Reply by oswaldB** [reply]
just find it is automatic.

**Reply by mjashanks** [reply]
Yes. When you create new apps, you can set the URL. We'll soon add the ability to edit URLs of existing apps too.

**Reply by mjashanks** [reply]
Actually - you can do this now - your app has to be Unpublished first. Click "Edit" in the menu below...

![image](https://user-images.githubusercontent.com/3524181/155103967-622daa37-d22f-4398-a42f-2611ddd2bc9d.png)

**Comment by SantoshSrinivas79** [comment]
Hello @mjashanks .. how can I check the version I am running? The version number doesn't show in the updates page now

![Screenshot 2022-02-23 at 1 08 29 PM](https://user-images.githubusercontent.com/1036163/155278291-83ac27a5-1565-42e9-aa9e-67cac5ee4ded.png)
!
